build/arm/linker/linker_nrf52840_bootloader.ld

```diff
@@ -7,17 +7,7 @@ __Stack_Init = _estack - __Stack_Size;
 _Minimum_Stack_Size = 0x1400;
 
 /* Memory Spaces Definitions */
-
-/* INCLUDE module_user_memory.ld */
-
-/* Ensure RAM region doesn't overlap with the system modules, since the bootloader imports
-   dynalib functions dynamically on certain platforms */
-/* bootloader_ram_length = 16K; */
-/* bootloader_ram_offset = 8K; */
-/* bootloader_ram_origin = user_module_sram_origin + bootloader_ram_offset; */
-/* ASSERT(bootloader_ram_offset + bootloader_ram_length <= user_module_sram_length, */
-/*     "Insufficient space for RAM region"); */
-
+INCLUDE platform_ram.ld
 INCLUDE softdevice.ld
 
 bootloader_ram_origin = 0x20000000 + APP_RAM_BASE;
```

Comment:
Should we define this in `platform_ram.ld` ?

Reply:
This PR addresses only modular builds of the main firmware to narrow the scope a little bit and keep the changes focused on compatibility with recent versions of GCC. I'd be happy to refactor linker files for the bootloader and monolithic builds as well (and also define flash memory maps in a similar manner), but not in the scope of this PR